GeneratorInterface/Hydjet2Interface/src/Hydjet2Hadronizer.cc

```diff
@@ -490,10 +499,12 @@ HepMC::GenVertex *Hydjet2Hadronizer::build_hyjet2_vertex(int i, int id) {
   // build verteces for the hyjets stored events
   double x0 = (hj2->GetX()).at(i);
   double y0 = (hj2->GetY()).at(i);
-  double x = x0 * cosphi0_ - y0 * sinphi0_;
-  double y = y0 * cosphi0_ + x0 * sinphi0_;
-  double z = (hj2->GetZ()).at(i);
-  double t = (hj2->GetT()).at(i);
+
+  // convert to mm (as in PYTHIA6)
+  double x = 0.000000000001 * (x0 * cosphi0_ - y0 * sinphi0_);
```

Comment:
It is less error prone, and more intuitive, if you define the conversion in a contexpr parameter

Reply:
How it is possible? We do not know the x0 and y0 at compile time...

Reply:
e.g.
```C++
  double constexpr am_to_mm = 0.000000000001;  // or even 1e-12, which is more intuitive
  double x =  am_to_mm * (x0 * cosphi0_ - y0 * sinphi0_);
```

Reply:
Ah, I see. But what is the reason of constexpr here? Why not just:
```
const double fm_to_mm = 1e-12; 
double x =  fm_to_mm * (x0 * cosphi0_ - y0 * sinphi0_);
```
?

Reply:
> Ah, I see. But what is the reason of constexpr here? Why not just:
> 
> ```
> const double fm_to_mm = 1e-12; 
> double x =  fm_to_mm * (x0 * cosphi0_ - y0 * sinphi0_);
> ```
> 
> ?

Well, `constexpr` is initialized at compile time, while `const` it is at runtime. Not a big difference here, let it as you updated it.